### DISCUSSION TITLE
Shared files to user root directory

### DISCUSSION THREAD

**Discussion by rashmimaharana** [drakkan/sftpgo#550] [Q&A] [answered]
Hi @drakkan and Team,

I have a requirement to share the files (not folder) from user "A" to user "B" root directory. So when user B logged in, it will get all shared files on its root directory. So no need to see folder mounted in its root directory and no need to dig into the folder to get files. 
Is that possible ?

All i can see the virtual folder(shared) mounting to another user account.

Appreciate for all your response :)

![image](https://user-images.githubusercontent.com/10846216/134206973-058d266d-32a9-4d3d-8750-807237c3b98c.png)

**Comment by drakkan** [comment]
try to set the same home dir for user "A" and user "B", if both users can write this will break quota

**Reply by rashmimaharana** [reply]
But in that case they both will see all files. I would like to have user "B" its own home directory and some files shared by user "A" to user "B".

**Reply by drakkan** [reply]
This is not possible sorry. If the user "B" is an external user I plan to add a sharing feature (via a web link, multiple files will be shared as zip) in future updates (no definite schedule), otherwise you are out of luck for now. 

If you need a particular feature you would consider to sponsor the project

https://github.com/sponsors/drakkan

but please let's discuss it first, this is an open source project, i don't accept or add anything just because someone wants to pay for it, I must be convinced that the feature improves SFTPGo, thanks

**Comment by drakkan** [comment, marked as answer]
> Hi @drakkan and Team,
> 
> I have a requirement to share the files (not folder) from user "A" to user "B" root directory. So when user B logged in, it will get all shared files on its root directory. So no need to see folder mounted in its root directory and no need to dig into the folder to get files. Is that possible ?
> 
> All i can see the virtual folder(shared) mounting to another user account.
> 
> Appreciate for all your response :)
> 
> ![image](https://user-images.githubusercontent.com/10846216/134206973-058d266d-32a9-4d3d-8750-807237c3b98c.png)

sharing is now supported

https://github.com/drakkan/sftpgo/commit/3bc58f5988fcfef970fd080a7d73716488f253ab

maybe it could help for this use case

**Reply by rashmimaharana** [reply]
Thanks @drakkan for this feature. You are a gem of a person. Really appreciate all your efforts and help :)

**Reply by rashmimaharana** [reply]
@drakkan 
I could see after sharing the files, one can download from the provided shared file link as zip. 
Is it possible to share the files to another user home directory or that is not included. Did I miss anything ?

**Reply by drakkan** [reply]
no sorry, this feature allows to share files with external users. To share with another user you can setup a shared folder, but the whole folder will be shared you cannot share just a file

**Reply by rashmimaharana** [reply]
Thanks @drakkan for the clarification and all support.
